test/e2e/serial/tests/workload_unschedulable.go

```diff
@@ -220,4 +220,133 @@ var _ = Describe("[serial][disruptive][scheduler] numaresources workload unsched
+					Expect(err).NotTo(HaveOccurred(), "unable to create pod %q on zone %q", podName, zone.Name)
+
+					paddingPods = append(paddingPods, padPod)
+				}
+			}
+
+			By("Padding target node")
+			// Pad the zones so no one could handle both containers
+			paddingForTagetNode := corev1.ResourceList{
+				corev1.ResourceCPU:    resource.MustParse("6"),
+				corev1.ResourceMemory: resource.MustParse("6Gi"),
+			}
+
+			targetNrt, err := e2enrt.FindFromList(nrtCandidates, targetNodeName)
+			Expect(err).NotTo(HaveOccurred(), "missing NRT info for targetNode %q", targetNodeName)
+			for idx, zone := range targetNrt.Zones {
+				podName := fmt.Sprintf("padding%s-%d", targetNodeName, idx)
+				padPod, err := makePaddingPod(fxt.Namespace.Name, podName, zone, paddingForTagetNode)
+				Expect(err).NotTo(HaveOccurred(), "unable to create padding pod %q on zone %q", podName, zone.Name)
+
+				padPod, err = pinPodTo(padPod, targetNodeName, zone.Name)
+				Expect(err).NotTo(HaveOccurred(), "unable to pin pod %q to zone %q", podName, zone.Name)
+
+				err = fxt.Client.Create(context.TODO(), padPod)
+				Expect(err).NotTo(HaveOccurred(), "unable to create pod %q on zone %q", podName, zone.Name)
+
+				paddingPods = append(paddingPods, padPod)
+			}
+
+			By("waiting for padding pods to be ready ...")
+			// wait for all padding pods to be up&running ( or fail)
+			failedPods := e2ewait.ForPodListAllRunning(fxt.Client, paddingPods)
+			for _, failedPod := range failedPods {
+				// no need to check for errors here
+				_ = objects.LogEventsForPod(fxt.K8sClient, failedPod.Namespace, failedPod.Name)
+			}
+			Expect(failedPods).To(BeEmpty(), "some padding pods have failed to run")
+
+			By("Scheduling the testing pod")
+			pod := objects.NewTestPodPause(fxt.Namespace.Name, "testpod")
+			pod.Spec.SchedulerName = serialconfig.Config.SchedulerName
+			pod.Spec.Containers = append(pod.Spec.Containers, pod.Spec.Containers[0])
+			pod.Spec.Containers[0].Name = pod.Name + "-cnt-0"
+			pod.Spec.Containers[0].Resources.Limits = requiredResCnt1
+			pod.Spec.Containers[1].Name = pod.Name + "cnt-1"
+			pod.Spec.Containers[1].Resources.Limits = requiredResCnt2
+
+			err = fxt.Client.Create(context.TODO(), pod)
+			Expect(err).NotTo(HaveOccurred(), "unable to create pod %q", pod.Name)
+
+			interval := 10 * time.Second
+			By(fmt.Sprintf("Checking pod %q keeps in %q state for at least %v seconds ...", pod.Name, string(corev1.PodPending), interval*3))
+			err = e2ewait.WhileInPodPhase(fxt.Client, pod.Namespace, pod.Name, corev1.PodPending, interval, 3)
+			if err != nil {
+				_ = objects.LogEventsForPod(fxt.K8sClient, pod.Namespace, pod.Name)
+			}
+			Expect(err).ToNot(HaveOccurred())
+
+			By(fmt.Sprintf("checking the pod was scheduled with the topology aware scheduler %q", serialconfig.Config.SchedulerName))
+			isFailed, err := nrosched.CheckPODSchedulingFailed(fxt.K8sClient, pod.Namespace, pod.Name, serialconfig.Config.SchedulerName)
```

Comment:
we should probably refine this function to have a stricter check (ofc in another PR)

Reply:
here: https://github.com/openshift-kni/numaresources-operator/pull/251